static/src/js/components/Errors/NotFound.js

```diff
@@ -1,59 +1,57 @@
-import React, { Component } from 'react'
+import React, { memo, useEffect } from 'react'
 import { v4 as uuidv4 } from 'uuid'
 
 import { eventEmitter } from '../../events/events'
 import LoggerRequest from '../../util/request/loggerRequest'
 import { locationPropType } from '../../util/propTypes/location'
 
-class NotFound extends Component {
-  UNSAFE_componentWillMount() {
+export const NotFound = memo(({
+  location
+}) => {
+  useEffect(() => {
```

Comment:
Does this useEffect need a dependency array?

Reply:
technically no, since it doesn't need re-rendering but I'll add it for consistency

Reply:
I think it does need an empty dependency array to make sure it only renders once, when the component mounts.